examples/lock-app/linux/src/LockAppCommandDelegate.cpp

```diff
@@ -0,0 +1,152 @@
+    const auto & params = self->mCommandParameters;
+    // Determine the endpoint ID from the parameters JSON. If it is missing, use the default endpoint defined in the
+    // door-lock-server.h
+    chip::EndpointId endpointId = DOOR_LOCK_SERVER_ENDPOINT;
+    if (params.isMember("EndpointId"))
+    {
+        VerifyOrExit(params["EndpointId"].isUInt(),
+                     ChipLogError(NotSpecified, "Lock App: Unable to execute command \"%s\": invalid endpoint Id",
+                                  self->mCommandName.c_str()));
+        endpointId = static_cast<chip::EndpointId>(params["EndpointId"].asUInt());
+    }
+
+    // TODO: Make commands separate objects derived from some base class to clean up this mess.
+
+    // Now we can try to execute a command
+    if (self->mCommandName == "SetDoorState")
+    {
+        VerifyOrExit(params.isMember("DoorState"),
+                     ChipLogError(NotSpecified,
+                                  "Lock App: Unable to execute command to set the door state: DoorState is missing in command"));
+
+        VerifyOrExit(
+            params["DoorState"].isUInt(),
+            ChipLogError(NotSpecified, "Lock App: Unable to execute command to set the door state: invalid type for DoorState"));
+
+        auto doorState = params["DoorState"].asUInt();
+        VerifyOrExit(doorState < to_underlying(DlDoorState::kUnknownEnumValue),
+                     ChipLogError(NotSpecified,
+                                  "Lock App: Unable to execute command to set door state: DoorState is out of range [doorState=%u]",
+                                  doorState));
+        LockManager::Instance().SetDoorState(endpointId, static_cast<DlDoorState>(doorState));
+    }
+    else if (self->mCommandName == "SendDoorLockAlarm")
+    {
+        VerifyOrExit(
+            params.isMember("AlarmCode"),
+            ChipLogError(NotSpecified, "Lock App: Unable to execute command to send lock alarm: AlarmCode is missing in command"));
+
+        VerifyOrExit(
+            params["AlarmCode"].isUInt(),
+            ChipLogError(NotSpecified, "Lock App: Unable to execute command to send lock alarm: invalid type for AlarmCode"));
+
+        auto alarmCode = params["AlarmCode"].asUInt();
+        VerifyOrExit(
+            alarmCode < to_underlying(DlAlarmCode::kUnknownEnumValue),
+            ChipLogError(NotSpecified,
+                         "Lock App: Unable to execute command to send lock alarm: AlarmCode is out of range [alarmCode=%u]",
+                         alarmCode));
+        LockManager::Instance().SendLockAlarm(endpointId, static_cast<DlAlarmCode>(alarmCode));
+    }
+    else
+    {
+        ChipLogError(NotSpecified, "Lock App: Unable to execute command \"%s\": command not supported", self->mCommandName.c_str());
+    }
+
+exit:
+    delete self;
+}
+
+void LockAppCommandDelegate::OnEventCommandReceived(const char * json)
+{
+    auto handler = LockAppCommandHandler::FromJSON(json);
```

Comment:
I was wondering if we can just move following parameters to LockAppCommandDelegate

    std::string mCommandName;
    Json::Value mCommandParameters;
    
 So that we don't have to create and copy those parameters to heap

Reply:
I'm a bit worried about thread safety in that case. It is entirely possible that we can overwrite command parameters when we've got a new command over the pipe while the other one is scheduled to work in SDK thread. As far as I understand, it's not guaranteed that the SDK thread and NamedPipe delegate are ran in the same thread. If there is such guarantee then I can change that to just a field in the delegate.

Reply:
That is a good argument, they are not running in the same thread, and could be a overwrite.